src/WebJobs.Script/Diagnostics/HostErrorLoggerProvider.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    public class HostErrorLoggerProvider : ILoggerProvider
```

Comment:
Do we need so many specialized logger providers? If that's the recommended design pattern then so be it, it just seems to me that we have a lot of them :)

Reply:
It should be about the same as TraceWriters. I tried to keep them 1:1. A logger is a stream of messages and when you want to route a message somewhere special, you add a loggerprovider. We want messages that go to:
- Host logs (all messages go here)
- Individual Function logs (only function-specific messages go here)
- App Insights
- System logs
- The HostErrorHandler (which we may be able to factor away but I wanted to keep it for now)